src/w2grid.js

```diff
@@ -474,7 +474,7 @@
                     }
                 }
                 if (match && returnIndex !== true) recs.push(this.records[i].recid);
-                if (match && returnIndex === true) recs.push(i);
+                if (match && returnIndex === true) recs.push(this.records[i]);
```

Comment:
Not sure it is right. If returnIntex is true, then it should return index of the record that is found.